### ISSUE TITLE
Slide not doing anything

### ISSUE THREAD

**Issue by Florberger** [paulmach/slide#13]
Hi

I'm new to this (GIT) so please tell me if I don't follow the practices/etiquette. 

I've been trying to use Strava Slide to update trails in Open Street Map - there is a Slide plugin on this page: https://strava.github.io/iD/#background=Bing&id=w54292996&map=17.60/11.74480/57.71357, that should be possible to use to update OSM - when using it, nothing happens though.

I've also tried the demo: https://labs.strava.com/slide/demo.html#16/-122.54277/38.05392, though there's no heat data being shown and when trying anyway, you get a server error message.

Is Slide "down" or is the problem on my side?

**Comment by paulmach** [comment]
yeah, everything is kind of broken ever since strava updated the heatmap. I will try to contact some people I still know there to see if they are interested in maintain it.

**Comment by EwenH** [comment]
@Florberger, what you wrote is exactly whats required with explanation and URL's. 

Paul, this has had significant impact on a lot of OSM editors on both verifying tracks and aligning. Please ask your friends at Strava however is there anyway we can surgically remove the code base off Strava and just consume the end point or better still, embed itstraight into the osm Id editor?

**Comment by Florberger** [comment]
@EwenH: Thanks for the feedback : )

@paulmach: Thanks for the reply - it's an awesome functionality you've created, it would be sad to see it disappear. Thanks for your great work!

**Comment by valleyofdawn** [comment]
This tool has revolutionized single-track OSM coverage in my area. Would be such a shame to see it go. Please let us know if we can do anything to see it rejuvenated. Petition Strava? Fork it somewhere else? Anything really.

**Comment by pdxvr** [comment]
If we can't get it to slide due to some reason or other, just having the heatmap as a background layer would make a huge difference.

**Comment by solitone** [comment]
That would be a fantastic tool! A JOSM plugin doing such thing would be extremely useful.

**Comment by nyeates** [comment]
Agreed that the heatmaps and slide REVOLUTIONIZED single-track trail
mapping on OSM.

Likely it contributed to 1000s of miles of new / improved mapped trails. I
updated two hundred alone and multiply that by hundreds more and you get
thousands of miles of added trails.

This means less people getting lost - search and rescue use OSM trail maps
(I saw them in a command center). It has a HUGE impact.

There are supposed to be workaround ways to get Strava heatmaps to load in
things like "JOSM" or "Go Map!!". A cleaner officially-supported way to get
to heatmaps for enthusiasts would be nice.

I think heatmaps access in apps currently involves getting a dev token from
Strava and utilizing that when loading it into a third party editor. I
couldn't get it to work. It is because Strava now requires you to LOGIN to
get at zoomed-in anonymous heatmaps data. All of this was likely because of
hidden military-bases being shown on maps via unknowing soldiers running
laps around their bases. Made news a year or two back and Strava had to
reign back heatmaps access. Heatmaps are still around though and appears to
continue to be updated.

**Comment by solitone** [comment]
I regularly use high resolution Strava Heatmap data in JOSM. All I need are the cookies that my browser gets from the Strava server, and set the imagery URL accordingly, like this:

`tms[3,15]:https://heatmap-external-{switch:a,b,c}.strava.com/tiles-auth/run/bluered/{zoom}/{x}/{y}.png?Key-Pair-Id=<MY_KEY_PAIR_ID_COOKIE_VALUE>&Policy=<MY_POLICY_COOKIE_VALUE>&Signature=<MY_SIGNATURE_COOKIE_VALUE>
`

I edit ways manually though, while this Slide utility would be a tremendous accelerator!

**Comment by nyeates** [comment]
I dislike JOSM. I see why it's liked by some but I'm not a super power
user. Mostly I don't get to edits unless I do them in the field. I'll
record GPS and then never get to it. Love the app 'Go Map!!' but I think
it's unsupported now. It does custom tile layers but no idea how to bring
cookies. Is there an explain anywhere on how to at least use the cookies
and set it up in JOSM? I couldn't figure it out last time

**Comment by pdxvr** [comment]
I am a power user but I can't stand JOSM.  Even if I took the time to add the hi res heat map to any editor view, the real value in this was the actual slide editor tool.  I corrected likely over 1000 miles of trails and roads just in my region.

**Comment by solitone** [comment]
Yes, the Slide tool would be really fantastic! Unfortunately I didn't know it and never tried it when was working.

@nyeates 
As for hi-res Strava imagery in JOSM, follow instruction steps 1-4 here:
https://github.com/solitone/josm_strava_cookies 

Then you should manually edit the imagery URL, changing it to something like:
`tms[3,15]:https://heatmap-external-{switch:a,b,c}.strava.com/tiles-auth/run/bluered/{zoom}/{x}/{y}.png?Key-Pair-Id=<YOUR_KEY_PAIR_ID_COOKIE_VALUE>&Policy=<YOUR_POLICY_COOKIE_VALUE>&Signature=<YOUR_SIGNATURE_COOKIE_VALUE>`

<YOUR_KEY_PAIR_ID_COOKIE_VALUE>, <YOUR_POLICY_COOKIE_VALUE>, and <YOUR_SIGNATURE_COOKIE_VALUE> are the values of the following 3 Strava cookies:
CloudFront-Key-Pair-Id, CloudFront-Policy, and CloudFront-Signature. How to get them depends on the browser you use. 

If you use Safari on macOS you could try the script that I wrote, following all those instruction points.

**Comment by RobJN** [comment]
Permission to continue using Strava Heatmap in OSM reconfirmed. There had been some doubt inthe OSM community after this tool stopped working.

https://lists.openstreetmap.org/pipermail/talk/2019-November/083564.html

Also there is a suggestion that the tool may be working again now. I've not yet tested.

**Comment by rene78** [comment]
> yeah, everything is kind of broken ever since strava updated the heatmap. I will try to contact some people I still know there to see if they are interested in maintain it.

Kudos to @paulmach ! Still sad that Slide is not working anymore. One of the most useful little helpers that I used for OSM mapping 👍

**Comment by endolith** [comment]
@pdxvr 

> If we can't get it to slide due to some reason or other, just having the heatmap as a background layer would make a huge difference.

You can: 

1. https://addons.mozilla.org/en-US/firefox/addon/josm-strava-heatmap/
2. Make sure you are logged in to Strava and have entered a location in your settings https://www.strava.com/settings/profile 
![image](https://github.com/paulmach/slide/assets/58611/c05f4d07-7cc9-4705-be9b-0f3f10710fae)

3. https://www.strava.com/heatmap
4. ![image](https://github.com/paulmach/slide/assets/58611/cf4d8911-f19f-40ec-a430-85dccce3c5d1)
5. ![image](https://github.com/paulmach/slide/assets/58611/62270486-f2cf-4e41-b343-fd83855754bb)

Can Slide be resurrected on another site?  I was considering making something like it but someone pointed out this existing project to me.
